### PR TITLE
Surface troubleshooting code from IIS Int repo

### LINKED ISSUE #6083: Add the IIS Integration sample code to the 'Troubleshooting' topic
In discussion with @Rob_R on the [Host on IIS topic](https://docs.microsoft.com/aspnet/core/host-and-deploy/iis), it occurs that many devs would find the IIS sample *very* helpful for set-up testing and troubleshooting. Add the sample to both of these topics ... this move is going to save some major headaches.

Thanks to @Rob_R :guitar: for discussion leading to this idea.

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #6083 

[Internal Review Topic (Troubleshoot ASP.NET Core projects)](https://review.docs.microsoft.com/en-us/aspnet/core/test/troubleshoot?view=aspnetcore-1.0&branch=pr-en-us-9689)

* The app code from IIS Integration that reports request, connection, env vars is so useful, including outside of IIS scenarios, that imo it should be surfaced to readers. I've used it several times, and the community has enjoyed using it.
* Engineering: The main update to review here is at the bottom of the diff (updates to the last file): *test/troubleshoot.md*.
* In most spots, adding a link to the *test/troublshoot.md* topic is enough. In a few spots, I add a short section.
* UE is not a thrust of this, but I did fix up a few things as I went along.

### CODE REVIEW

File: aspnetcore/test/troubleshoot.md [hunk 1]
```diff
@@ -61,3 +61,88 @@ This warning appears when the environment variable `PATH` doesn't point to any .
 
 * Install or verify the .NET Core SDK is installed.
 * Verify that the `PATH` environment variable points to the location in which the SDK is installed. The installer normally sets the `PATH`.
+
+## Obtain data from an app
+
+If an app is capable of responding to requests, you can obtain the following data from the app using terminal inline middleware:
```
Comment (Tratcher) [inline aspnetcore/test/troubleshoot.md]:
```suggestion
If an app is capable of responding to requests, you can obtain the following data from the app using middleware:
```

File: aspnetcore/test/troubleshoot.md [hunk 2]
```diff
@@ -61,3 +61,88 @@ This warning appears when the environment variable `PATH` doesn't point to any .
...
+
+* Request &ndash; Method, scheme, host, pathbase, path, query string, headers
+* Connection &ndash; Remote IP address, remote port, local IP address, local port, client certificate
+* Identity &ndash; Name, display name
+* Environment variables
+
+Place the following [terminal middleware](xref:fundamentals/middleware/index#create-a-middleware-pipeline-with-iapplicationbuilder) code snippet at the start of the `Startup.Configure` method's request processing pipeline. The environment is checked before the middleware is run to ensure that the code is only executed in the Development environment.
```
Comment (Tratcher) [inline aspnetcore/test/troubleshoot.md]:
```suggestion
Place the following [middleware](xref:fundamentals/middleware/index#create-a-middleware-pipeline-with-iapplicationbuilder) code snippet at the start of the `Startup.Configure` method's request processing pipeline. The environment is checked before the middleware is run to ensure that the code is only executed in the Development environment.
```

File: aspnetcore/test/troubleshoot.md (hunk 1 above)
Comment (jkotalik) [inline aspnetcore/test/troubleshoot.md]: @Tratcher How useful do you feel this data is for troubleshooting?

File: aspnetcore/test/troubleshoot.md [hunk 3]
```diff
@@ -61,3 +61,88 @@ This warning appears when the environment variable `PATH` doesn't point to any .
...
+            foreach (var key in vars.Keys.Cast<string>().OrderBy(key => key, 
+                StringComparer.OrdinalIgnoreCase))
+            {
+                var value = vars[key];
+                sb.Append($"{key}: {value}{nl}");
+            }
+
+            context.Response.ContentType = "text/plain";
+            await context.Response.WriteAsync(sb.ToString());
+        });
```
Comment (Tratcher) [inline aspnetcore/test/troubleshoot.md]: Can you add config to this sample as well? People not loading the expected config files is a common issue.
https://github.com/aspnet/KestrelHttpServer/blob/5c1fcd664d39db8fe5c8e38052a3cc29f90322f6/samples/SampleApp/Startup.cs#L178-L185

File: aspnetcore/test/troubleshoot.md (hunk 1 above)
Comment (Tratcher) [reply aspnetcore/test/troubleshoot.md]: It depends on the issue. It's great for debugging forwarders, config, etc..

**Comment by guardrex** [comment]
@jkotalik @Tratcher Made your suggested updates. Ready for :eyes:.

File: aspnetcore/test/troubleshoot.md [hunk 4]
```diff
@@ -61,3 +61,97 @@ This warning appears when the environment variable `PATH` doesn't point to any .
...
+            }
+            else
+            {
+                sb.Append($"Status: Disabled{nl}{nl}");
+            }
+
+            sb.Append($"Configuration{rule}");
+            foreach (var pair in config.GetChildren())
+            {
+                sb.Append($"{pair.Path}: {pair.Value}{nl}");
```
Comment (Tratcher) [inline aspnetcore/test/troubleshoot.md]: This only shows top level keys. The sample I showed you was intentionally recursive:
https://github.com/aspnet/KestrelHttpServer/blob/5c1fcd664d39db8fe5c8e38052a3cc29f90322f6/samples/SampleApp/Startup.cs#L178-L185

**Comment by guardrex** [comment]
@Tratcher How about this approach ... this is what I normally use ...

https://github.com/aspnet/Docs/pull/9689/commits/426ee7dff4cbdec71e00e3de7ee180721e3a144d

File: aspnetcore/test/troubleshoot.md [hunk 5]
```diff
@@ -61,3 +61,97 @@ This warning appears when the environment variable `PATH` doesn't point to any .
...
+
+* Request &ndash; Method, scheme, host, pathbase, path, query string, headers
+* Connection &ndash; Remote IP address, remote port, local IP address, local port, client certificate
+* Identity &ndash; Name, display name
+* Configuration settings
+* Environment variables
+
+Place the following [middleware](xref:fundamentals/middleware/index#create-a-middleware-pipeline-with-iapplicationbuilder) code at the beginning of the `Startup.Configure` method's request processing pipeline. The environment is checked before the middleware is run to ensure that the code is only executed in the Development environment.
+
+To obtain the environment use either of the following approaches:
```
[the remark this answers is not in the document]
Comment (guardrex) [reply aspnetcore/test/troubleshoot.md]: ooooooops!!! :smile:

**Pull request merged**